java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -205,9 +210,11 @@ public AccessToken refreshAccessToken() throws IOException {
             }
```

Comment:
possibly everything above here should be moved, as it's not actually refreshing the token.

Reply:
and actually, this is `refreshAccessTokenIfNeeded` or something - as it won't *necessarily* refresh.